lib/syskit/roby_app/plugin.rb

```diff
@@ -727,16 +727,27 @@ def using_ros_launcher(name, options = {})
 
             # Start all deployments
             #
-            # @param [String,nil] on the name of the process server on which
-            #   deployments should be started. If nil, all servers are considered
-            def syskit_start_all_deployments(on: nil, except_on: "unmanaged_tasks")
+            # @param [String] on the name of the process servers on which
+            #   deployments should be started. If empty, all servers are considered
+            # @param [String] except_on the of the process servers on which deployments
+            #   should NOT be started.
+            def syskit_start_all_deployments(
+                deployment_group: Syskit.conf.deployment_group,
+                on: [],
+                except_on: ["unmanaged_tasks"]
+            )
                 existing_deployments = plan.find_tasks(Syskit::Deployment)
                                            .not_finished
                                            .find_all(&:reusable?)
                                            .map(&:process_name).to_set
 
-                Syskit.conf.each_configured_deployment(on: on, except_on: except_on) do |configured_deployment|
-                    next if existing_deployments.include?(configured_deployment.process_name)
+                deployment_group.each_configured_deployment(
+                          on: on,
+                          except_on: except_on
+                ) do |configured_deployment|
```

Comment:
I tried recreating the interface as it was in the previous version, but I am not sure if it makes sense to filter for the process server name on the `deployment_group` side, can a deployment group have more than one process server name? If not, filtering for `on` and `except_on` could be done inside this method, leaving `deployment_group.each_configured_deployment` unchanged.

Reply:
It does make sense generally speaking ... but also find it weird. It feels more flexible and not worse to do ad-hoc filtering in each method that need it, instead of having an inherently limited interface in each_configured_deployment.